### PR TITLE
Redesign.

### PR DESCRIPTION
**Pull request by ktustanowski**

### CODE REVIEW

File: UnitTestApplicationTests/LoginViewControllerTests.swift [hunk 1]
```diff
@@ -35,106 +35,40 @@ class LoginViewControllerTests: XCTestCase {
...
-        viewController.viewDidLoad()
-        viewController.loginValidationLabelHeightConstraint.constant = 999
-        
-        viewController.loginTextField.validators?.last?.isValidAction?()
-        
-        XCTAssertEqual(viewController.loginValidationLabelHeightConstraint.constant, 0, "Is empty validator should hide validation label on success")
-    }
     
-    func testThatLoginIsEmailValidatorIsInvalidActionIsSetToShowValidationLabelWithValidationText() {
+    func testThatLoginIsEmailValidatorIsSetOnViewDidLoad() {
```
Comment (dbacinski) [inline UnitTestApplicationTests/LoginViewControllerTests.swift]: test name should be reflected in assertions, I would give a better name, we don't care if validator is set or not but if controller gives expected result.

File: UnitTestApplicationTests/LoginViewControllerTests.swift [hunk 2]
```diff
@@ -183,6 +117,20 @@ class LoginViewControllerTests: XCTestCase {
         XCTAssertEqual(viewController.spinner.isAnimating, true)
     }
     
+    func testThatBeforeSigningInUserProviderIsCreated() {
+        //Given
+        viewController.loginTextField.text = "kyle.katarn@ravensclaw.com"
+        viewController.passwordTextField.text = "0123456"
+        viewController.successLabel.text = nil
+        viewController.signInButton.isHidden = false
+        viewController.spinner.stopAnimating()
```
Comment (dbacinski) [inline UnitTestApplicationTests/LoginViewControllerTests.swift]: i guess that above attributes are not parameters of this test, I would extract them into method setupInitialState

Comment (dbacinski) [reply UnitTestApplicationTests/LoginViewControllerTests.swift]: just extract parameters that are not relevant for this test

File: UnitTestApplicationTests/LoginViewControllerTests.swift [hunk 3]
```diff
@@ -211,11 +158,13 @@ class LoginViewControllerTests: XCTestCase {
...
         viewController.signIn()
         //Then
-        XCTAssertEqual(viewController.successLabel.text, "Couldn't login 😱", "Should contain \"Couldn't login 😱\"")
+        XCTAssertEqual(viewController.successLabel.text, "Couldn't login 😱", "Should be equal Couldn't login 😱")
         XCTAssertEqual(viewController.signInButton.isHidden, false)
         XCTAssertEqual(viewController.spinner.isAnimating, false)
     }
 
+    
+
```
Comment (dbacinski) [inline UnitTestApplicationTests/LoginViewControllerTests.swift]: remove redundant new lines

**Review by dbacinski — CHANGES_REQUESTED** [review]

File: UnitTestApplicationTests/LoginViewControllerTests.swift (hunk 1 above)
Comment (ktustanowski) [reply UnitTestApplicationTests/LoginViewControllerTests.swift]: I was thinking about this and in this case I actually want to make sure that all validators are set and present - and the way I'm checking it is by calling sign in (thanks to your advice). This assertions are saying - this is set. If this is not enormous flaw in your opinion I would leave this as is.

**Comment by dbacinski** [comment]
+1 looks good to me

File: UnitTestApplicationTests/LoginViewControllerTests.swift (hunk 1 above)
Comment (dbacinski) [reply UnitTestApplicationTests/LoginViewControllerTests.swift]: ok

**Pull request merged**